ruff_lsp/server.py

```diff
@@ -416,9 +431,50 @@ def code_action(params: CodeActionParams) -> list[CodeAction] | None:
                             title=title,
                             kind=CodeActionKind.QuickFix,
                             data=params.text_document.uri,
-                            edit=_create_workspace_edit(
-                                document, cast(Fix, diagnostic.data)
-                            ),
+                            edit=_create_workspace_edit(document, fix),
+                            diagnostics=[diagnostic],
+                        ),
+                    )
+
+    # Add "Disable for this line" for every diagnostic.
+    if not params.context.only or CodeActionKind.QuickFix in params.context.only:
+        for diagnostic in params.context.diagnostics:
+            if diagnostic.source == "Ruff":
+                noqa_row = cast(DiagnosticData, diagnostic.data)["noqa_row"]
```

Comment:
I think this needs to be more defensive, since this key could be absent for older versions of Ruff (as opposed to being `None`).

Reply:
The DiagnosticData is created by the LSP, when I create it I use `noqa_row=check.get("noqa_row")` so I think it's OK?

Reply:
Ah you're right!